### ISSUE TITLE
Autoencoder with 0 nodes learns something

### ISSUE THREAD

**Issue by NilsWinter** [fchollet/keras#3721]
Hi all,

I am currently running some test with simple Autoencoders. I just copied and pasted the code from this keras blog entry: https://blog.keras.io/building-autoencoders-in-keras.html

However, when I was testing different architectures, I just found out that even an Autoencoder with zero nodes in the (well, technically not even existing) first layer appears to be learning something. The loss I get is comparable with the loss I get with bigger architectures.

Could this be a bug in the Keras autoencoder code or might this be a problem of my dataset (which is quite noisy). My intuition was that I shouldn't learn anything when using a layer with zero nodes.

Any suggestions would be very helpful!
Thanks a lot!
